src/lambda_toolshed/papillon.cljc

```diff
@@ -120,11 +117,29 @@
     result))
 
 (defn- present-async
-  [result]
-  (go-loop [ctx result]
-    (if (satisfies? ReadPort ctx)
-      (recur (<! ctx))
-      (or (::error ctx) ctx))))
+  [result presenter]
+  (if (async? result)
+    (emerge result #(present-async % presenter))
+    (presenter result)))
+
+#?(:cljs
```

Comment:
My idea was that it would allow someone else to give a different presenter (e.g. manifold or promesa), or pick if they want promises vs channels if in CLJS.

Reply:
Would `present-as-promise` and `present-as-channel` be better names?

Reply:
is there a way to achieve both? unification of terms, while maintaining optionality/composability? possibly protocol driven?

Reply:
Yep, my belief/hope is that the terms can be unified.  It's not a question of naming -it's the generalization (or lack thereof).